### PR TITLE
Tweak to sidestaking to ensure stake return output matches

### PR DESCRIPTION
**Pull request by jamescowens**
input exactly in the corner case where the sidestake distributions
add up to 100%.

### CODE REVIEW

**Comment by jamescowens** [comment]
This can also be applied cleanly (and should) to the development branch.

**Comment by jamescowens** [comment]
Thomas. Nethlek and I have tested this very small tweak and it works. I want to get it merged into Camilla for general testing before Camilla is released.

File: src/miner.cpp [hunk 1]
```diff
@@ -718,18 +721,33 @@ void SplitCoinStakeOutput(CBlock &blocknew, int64_t &nReward, bool &fEnableStake
...
-            blocknew.vtx[1].vout.push_back(CTxOut(nReward * iterSideStake->second, SideStakeScriptPubKey));
+
+            int64_t nSideStake = 0;
+
+            // For allocations ending less than 100% assign using sidestake allocation.
+            if (dSumAllocation + iterSideStake->second < 1.0)
+                nSideStake = nReward * iterSideStake->second;
+            // We need to handle the final sidestake differently in the case it brings the total allocation up to 100%,
+            // because testing showed in corner cases the output return to the staking address could be off by one Halford.
+            else if (dSumAllocation + iterSideStake->second == 1.0)
```
Comment (denravonska) [inline src/miner.cpp:741]: Do we need to account for cases when the sum is >1.0 or is that handled elsewhere?

Edit: It will default to 0, right?

Comment (jamescowens) [reply src/miner.cpp:741]: That check has always been there — line 716.

Comment (denravonska) [reply src/miner.cpp:741]: Ah, there it is, hidden by the diff.

**Review by denravonska — APPROVED** [review]

**Pull request merged**